assets/dev/js/editor/components/ai/pages/form-code/code-block.js

```diff
@@ -0,0 +1,42 @@
+import { useRef } from 'react';
+import { Box, Button } from '@elementor/ui';
+import TextArea from '../../components/text-area';
+
+const CodeBlock = ( { node, inline, children, defaultValue, onInsert, ...props } ) => {
+	const codeBlockInput = useRef( null );
+
+	if ( inline ) {
+		return <code { ...props } />;
+	}
+
+	return (
+		<Box sx={ { position: 'relative' } }>
+			<TextArea
+				ref={ codeBlockInput }
+				defaultValue={ children[ 0 ] }
+				sx={ { mb: 3 } }
+				helperText={ __( 'Code generated by AI may be inaccurate.', 'elementor' ) }
+				{ ...props }
+			/>
+
+			<Button
+				size="small"
+				variant="contained"
+				onClick={ () => onInsert( defaultValue + '\n' + codeBlockInput.current.value ) }
```

Comment:
maybe call it prefix or something?

```suggestion
				onClick={ () => onInsert( prefix + '\n' + codeBlockInput.current.value ) }
```

Reply:
But it's not used as a prefix, it's the current code from the control..